### ISSUE TITLE
Standard Maven Repository

### ISSUE THREAD

**Issue by crazyash** [standard-webhooks/standard-webhooks#220]
Are there any plans to include the java library in standard maven repository?

**Comment by tasn** [comment]
Yes we should!

**Comment by crazyash** [comment]
> Yes we should!

Any timelines that you foresee @tasn ? We want to use it as a standard library for our Org..

**Comment by crazyash** [comment]
Also @tasn 

We see webhook-id doesn't contain `msg_` prefix in example mentioned in https://nordicapis.com/creating-industry-standard-webhooks-with-standard-webhooks/

```
POST /chaps-notifications

webhook-id: 2ace281c-ffed-478c-b61b-6dc6bd07e1cc
webhook-timestamp: 1703548799
webhook-signature: v1,K5oZfzN95Z9UVu1EsfQmfVNQhnkZ2pj9o9NDN/H/pI4= v1a,hnO3f9T8Ytu9HwrXslvumlUpqtNVqkhqw/enGzPCXe5BdqzCInXqYXFymVJaA7AZdpXwVLPo3mNl8EM+m7TBAg==

2ace281c-ffed-478c-b61b-6dc6bd07e1cc.1703548799.{ ... }
```

**Questions**
- What is the standard here? Should we use the prefix or not?
- Any specific purpose for having `msg_` as a prefix?

**Comment by tasn** [comment]
msg_ prefix is not required there.

Timeline: let me try taking a look at it today! Worst case next week.

**Comment by tasn** [comment]
Quick update: waiting on maven central to approve the namespace.

**Comment by tasn** [comment]
Looks like it's up and working! Can you please confirm?
```
pkg:maven/com.standardwebhooks/standardwebhooks@1.0.0
```

**Comment by crazyash** [comment]
@tasn It's only 2KB.. I think necessary jars are not included. Is it possible to make a fat jar?

**Comment by tasn** [comment]
We don't have any external dependencies, what additional jars are you looking for?

Anyhow, this is a bit beyond my Java expertise. Mind helping? The source code is here: https://github.com/standard-webhooks/standard-webhooks/tree/main/libraries/java

**Comment by crazyash** [comment]
Jar contains only the META-INF folder and none of the classes..

Also, I see all the gradle files are deleted with the recent commit. There were below dependencies earlier..

**Refer:** https://github.com/standard-webhooks/standard-webhooks/blob/f3a49acf8447f0c7e1af1383eecfd9c24d49815d/libraries/java/build.gradle

```
dependencies {
    implementation 'io.swagger:swagger-annotations:1.6.12'
    implementation "com.google.code.findbugs:jsr305:3.0.2"
    implementation 'com.squareup.okhttp3:okhttp:4.12.0'
    implementation 'com.squareup.okhttp3:logging-interceptor:4.12.0'
    implementation 'com.google.code.gson:gson:2.10.1'
    implementation 'io.gsonfire:gson-fire:1.9.0'
    implementation group: 'org.apache.commons', name: 'commons-lang3', version: '3.17.0'
    implementation 'org.threeten:threetenbp:1.7.0'
    implementation 'javax.annotation:javax.annotation-api:1.3.2'
    testImplementation 'junit:junit:4.13.1'
}
```

Changes mentioned in https://github.com/standard-webhooks/standard-webhooks/issues/219 would have solved the issues.

**Comment by tasn** [comment]
These were mistakes, they weren't actually used / imported. Does the Jar not work because some error?

**Comment by crazyash** [comment]
jar file should contain the classes. But the one from maven repository contains only META-INF folder in it.

It should contain classes like below:
<img width="310" alt="image" src="https://github.com/user-attachments/assets/a8376c26-1569-4afe-8f3d-a718a13cb956" />

Here's the jar I created using gradle: [standardwebhooks-1.0.0.jar.zip](https://github.com/user-attachments/files/18317851/standardwebhooks-1.0.0.jar.zip)

**Comment by tasn** [comment]
Oh, so the release process wasn't actually good, gotcha. Would you mind taking a look at what I did wrong? I had to switch to Maven because of changes in Maven Central, and it could be that I got something wrong.

Your help would be greatly appreciated, as it sounds like you know your stuff, and I'm very rusty when it comes to Java.

**Comment by tasn** [comment]
Hey, just checking in on this!

**Comment by crazyash** [comment]
Hi,

I tried to push the fix to a branch here but couldn't.

I forked it and pushed the fixes. Here's the fixed code:
https://github.com/crazyash/standard-webhooks/tree/main

**Comment by tasn** [comment]
Unless I'm misreading this patch, this moves things back to gradle? We are on `maven` now (long story, maven central is a pain). Do you have any idea how to fix it with maven?

**Comment by tasn** [comment]
Are you talking about the jars of the dependencies? Maybe we are missing something like https://maven.apache.org/plugins/maven-shade-plugin/?

**Comment by crazyash** [comment]
Yes. It is Gradle. Initially when I forked, it was setup with gradle.
Unfortunately, I am not an expert in maven. So, can't help there.

**Comment by tasn** [comment]
That's my problem, I also don't know much about Maven. I'll try hitting their support again. :|

But can you describe the exact issue that you're facing? Is the problem that the jar has no classes, or that the jar doesn't have the dependencies bundled into the jar as well?

**Comment by crazyash** [comment]
As you said, no dependencies are used in the actual code.. They were added by mistake..

Issue is that jar doesn't have any class files.

**Comment by tasn** [comment]
OK, I'll ask someone for help.
